File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/OperationStatus.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

/**
 * Status of the operation.
 */
public enum OperationStatus {

    OK("OK"),
    NOK("NOK"),
    ERROR("ERROR"),
    PENDING("PENDING"),
    UNKNOWN("UNKNOWN");

    private String value;

    OperationStatus(String value) {
        this.value = value;
    }

    public static OperationStatus fromValue(String value) {
        for (OperationStatus b : OperationStatus.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: upl-purchase-sdk-php/src/Model/OperationStatus.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use InvalidArgumentException;

use function array_key_exists;
use function sprintf;

/**
 * Status of the operation.
 */
class OperationStatus implements Enum
{
    public const OK = 'OK';
    public const NOK = 'NOK';
    public const ERROR = 'ERROR';
    public const PENDING = 'PENDING';
    public const UNKNOWN = 'UNKNOWN';

    public const ALLOWED_TYPES = [
        'OK' => self::OK,
        'NOK' => self::NOK,
        'ERROR' => self::ERROR,
        'PENDING' => self::PENDING,
        'UNKNOWN' => self::UNKNOWN,
    ];

    /**
     * @var string
     */
    private $value;

    /**
      * @param string $value
      * @throws InvalidArgumentException
      */
    public function __construct(string $value)
    {
        if (!array_key_exists($value, self::ALLOWED_TYPES)) {
            throw new InvalidArgumentException(sprintf('Unexpected value "%s"', $value));
        }
        $this->value = $value;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->value;
    }
}
